### PR TITLE
Refractor

### PR DESCRIPTION
**Pull request by Cremba**
refractor the project to have src/... 
added wrapper for mono usage
added script to pack the project to an npm file

### CODE REVIEW

**Comment by A-Manning** [comment]
Please review the files being committed here.
In particular, please do not commit:
.paket/load
contract-templates.sln.DotSettings.user
src/.vs/contract-templates/xs/UserPrefs.xml

Also, there is no need to have paket generate .csx files.

**Comment by Cremba** [comment]
I added them to the gitignore

File: paket.dependencies [hunk 1]
```diff
@@ -1,11 +1,11 @@
 generate_load_scripts: true
 source https://www.nuget.org/api/v2
 
-framework: auto-detect
```
Comment (A-Manning) [inline paket.dependencies:4]: Will the build work with `auto-detect`? If this works, please don't change it

Comment (Cremba) [reply paket.dependencies:4]: solved

**Pull request merged**